webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -0,0 +1,760 @@
+
+    const providerContainer = document.createElement("hbox");
+    panelCloudNotification.appendChild(providerContainer);
+
+    const icon = document.createElement("image");
+    icon.id = "cloudDownloadTypeIcon";
+    providerContainer.appendChild(icon);
+
+    const cloudDownloadContainer = document.createElement("vbox");
+    cloudDownloadContainer.id = "cloudDownloadContainer";
+
+    const desc = document.createElement("description");
+    desc.id = "cloudDownloadTitle";
+    cloudDownloadContainer.appendChild(desc);
+
+    const hbox = document.createElement("hbox");
+    cloudDownloadContainer.appendChild(hbox);
+    const innerDesc = document.createElement("description");
+    innerDesc.id = "cloudDownloadDetail";
+    const prefLink = document.createElement("label");
+    prefLink.className = "text-link";
+    prefLink.id = "cloudDownloadPreference";
+    hbox.appendChild(innerDesc);
+    hbox.appendChild(prefLink);
+
+    const radiogroup = document.createElement("radiogroup");
+    radiogroup.id = "multiProviderSelect";
+    cloudDownloadContainer.appendChild(radiogroup);
+
+    providerContainer.appendChild(cloudDownloadContainer);
+
+    const buttonContainer = document.createElement("hbox");
+    panelCloudNotification.appendChild(buttonContainer);
+    for (const id of ["cloudDownloadCancel", "cloudDownloadSave"]) {
+      const button = document.createElement("button");
+      button.id = id;
+      button.className = "panelCloudNotificationUI-button";
+      buttonContainer.appendChild(button);
+    }
+    buttonContainer.lastChild.setAttribute("default", "true");
+    return fragment;
+  },
+
+  /**
+   * Display Cloud Storage Notification inside download panel that asks user to
+   * save all subsequent downloads to their preferred cloud storage provider by changing
+   * download settings
+   */
+
+  async showNotification() {
+    const browserWindow = this.getRecentWindow();
+
+    if (!browserWindow || !browserWindow.document) {
+      return;
+    }
+
+    const document = browserWindow.document;
+    const panelDownload = document.getElementById("downloadsPanel-mainView");
+    if (!panelDownload) {
+      return;
+    }
+
+    const providersMap = this.providers;
+
+    // Continue only if cloud providers exists on user device
+    if (providersMap.size === 0) {
+      return;
+    }
+
+    // Check if user had already opted-in or
+    // enough time has passed since last prompts shown
+    if (!await CloudStorage.promisePromptInfo()) {
+      return;
+    }
+
+    if (CloudDownloadsInternal.checkIfExistingCloudProviderDownloadSettings()) {
+      return;
+    }
+
+    // Before showing notification check if move download context menu was
+    // registered previously, exit if attempt to successfully register context menu fails
+    try {
+      const moveDownloadMenuItem = document.getElementById("moveDownload");
+      if (!moveDownloadMenuItem) {
+        await this.registerContextMenu(browserWindow);
```

Comment:
This shouldn't ever be true right?

Reply:
I failed to replicate the bug because of which I put this check, this shouldn't be needed, removing for now.